src/setup/config.rs

```diff
@@ -18,6 +23,56 @@ pub struct NodeConfig {
     pub log_to_stdout: bool,
     /// The path of the cache directory of the node.
     pub path: PathBuf,
+    /// The network socket address of the node.
+    pub local_addr: Option<SocketAddr>,
+    /// The REST API socket address of the node.
+    pub rest_api_addr: Option<SocketAddr>,
+}
+
+impl NodeConfig {
+    /// Continuously try to read a string from a file.
+    async fn try_read_to_string(file_path: &Path) -> String {
+        loop {
+            match fs::read_to_string(&file_path) {
+                Ok(content) => return content,
+                Err(_) => {
```

Comment:
I'm assuming the expected error here is a missing file, but how to make sure we read it in its entirety? we might need to apply parsing after the read

Reply:
> If parsing fails, aren't we doomed anyway? We can do nothing about the files the Algorand node has generated.

Yeah, _but_ if we attempt to parse an incomplete file and it fails, we'd still retry the file read, at which point the contents might be complete.

Reply:
That being said, I'd say the probability of us reading a file that hasn't fully been written is pretty low, considering its contents.